src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToGenerateImplicitNamespaceImports_DotNet.cs

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using FluentAssertions;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToGenerateImplicitNamespaceImports_DotNet : SdkTest
+    {
+        private const string _testProjectName = "TestConsoleApp";
+        private const string _importFileName = "ConsoleApp.ImplicitNamespaceImports.cs";
+
+        public GivenThatWeWantToGenerateImplicitNamespaceImports_DotNet(ITestOutputHelper log) : base(log) { }
+
+        [Fact]
+        public void It_generates_imports_and_builds_successfully()
+        {
+            var tfm = "net6.0";
+            var testAsset = _testAssetsManager
+                .CopyTestAsset(_testProjectName)
+                .WithSource()
+                .WithTargetFramework(tfm);
```

Comment:
Here's how to create the project in code via the `TestProject` APIs like I suggested earlier:

```suggestion
            var testProject = new TestProject
            {
                IsExe = true,
                TargetFrameworks = tfm,
            };
            testProject.AdditionalProperties["LangVersion"] = "preview";

            var testAsset = _testAssetsManager.CreateTestProject(testProject);
```

By default the test project name will be set automatically via `CallerMemberNameAttribute`, and you can access it later with the `Name` property in order to construct the expected import filename.